bear/main.py.in

```diff
@@ -296,9 +296,9 @@ def intercept_build():
     if args.append and os.path.isfile(args.cdb):
```

Comment:
Could you filter out the excluded files before calling the `CompilationDatabase.save` method? I would like to keep the save functionality as is. It can be a simple function like this:

```python
def exclude(candidates, directories):
     # type: (Iterable[Compilation], List[str]) -> Iterable[Compilation]

    def is_no_excluded(candidate):
         return not any(candidate.is_in(directory) for directory in directories)

    return (candidate for candidate in candidates if is_no_excluded(candidate))
```

This would require to create an `is_in` method on the `Compilation` class. Please use the `os.path` module to check if the directory contains the file or not. (String comparision might be misleading.)